### DISCUSSION TITLE
Using TipTap Richtext within Mantine useForm

### DISCUSSION THREAD

**Discussion by riccardolardi** [mantinedev/mantine#6577] [Q&A]
I'm looking for the best possible way to use the TipTap Richtext editor within `useForm` to take advantage of form validation and controlled input. Is this the right way to go about this or is there a more recommended way?

```tsx
export default function RichText({ ...props }: InputWrapperProps & TextareaProps) {
  const editor = useEditor({
    content: props.value as string,
    // @ts-expect-error ??? need to fix this
    onUpdate: ({ editor }) => props.onChange?.(editor.getHTML()),
  });

  return (
    <Input.Wrapper {...props}>
      <RichTextEditor
        editor={editor}
        style={ props.error ? { borderColor: "red" } : {} }
      >
        {/* ... */}
      </RichTextEditor>
    </Input.Wrapper>
  );
}
```

And in parent form component:
```tsx
<RichText label="TipTap" required { ...form.getInputProps("tiptap") } />
```

**Comment by rtivital** [comment]
What's the issue? Doesn't it work for you?

**Reply by riccardolardi** [reply]
I'm just wondering whether this is the right way to do this or if there is a better way? And also, how to type the props.onChange correctly - you see, I'm ignoring the TS error

**Reply by CrusaderGoT** [reply]
Did you find a better way to do it? Either way, thanks for your post, i was confused on how to approach the issue.
